components/permalink_view/permalink_view.jsx

```diff
@@ -47,8 +38,10 @@ export default class PermalinkView extends React.PureComponent {
         document.body.classList.add('app__body');
     }
 
-    componentDidUpdate() {
-        if (!this.state.valid) {
+    componentDidUpdate(prevProps) {
+        const prevPostid = prevProps.match.params.postid;
+        const currPostid = this.props.match.params.postid;
+        if (prevPostid !== currPostid) {
```

Comment:
I fail to see how the post id might change, I kept it only for the sake of discussion

Reply:
I think this would happen if you clicked on a few different permalinks in a short time period because I think it might reuse the PermalinkView between them.